### PR TITLE
Allow configurable sort options

### PR DESCRIPTION
**Pull request by peterjcoffman**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [ ] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by peterjcoffman** [comment]
I need to come up with a good toolbox branch to show off this feature and https://github.com/xh/hoist-react/pull/1696.

**Comment by peterjcoffman** [comment]
@haynesjm42 You initially proposed this change to go along with https://github.com/xh/hoist-react/issues/1636. It looks like we might not be merging the `maxRows` change, but what about this one?

File: cmp/grid/impl/ColumnHeader.js [hunk 1]
```diff
@@ -34,14 +34,13 @@ export const ColumnHeader = hoistCmp({
         const sortIcon = () => {
             const activeGridSorter = impl.activeGridSorter;
             if (!activeGridSorter) return null;
+            const {abs, sort} = activeGridSorter;
 
             let icon;
```
Comment (lbwexler) [inline cmp/grid/impl/ColumnHeader.js]: This seems like generally a good fix to appropriately support abs ascending, if it was ever desired by an application.

**Comment by lbwexler** [comment]
Tom -- can you review and merge this if it seems reasonable to you?

**Comment by lbwexler** [comment]
Note that as a useful bonus here, we are also enabling that shift-click will remove a sort when that column is the only column currently sorted.

Makes it easier to get back to natural sorting, and behaves similar to selection.

**Comment by TomTirapani** [comment]
Looks great!

I noticed that keeping track of `sortIndex` within the column meant that `getNextSortBy()` wasn't aware of programmatic changes to the sortBy (including default sortBy). I changed that logic there to derive the index from the current `activeGridSorter`.

**Comment by TomTirapani** [comment]
I'm going to go ahead and merge, but if anyone wants to check out my tweaks they're here: https://github.com/xh/hoist-react/pull/1695/commits/0df4439ab8320175a3fae683770d4d2d47f2b345

**Pull request merged**